optimade/validator/validator.py

```diff
@@ -460,8 +463,25 @@ def test_as_type(self):
         self.deserialize_response(response, self.as_type_cls)
 
     @test_case
-    def test_page_limit(self, response):
-        """ Test that a multi-entry endpoint obeys the page limit. """
+    def test_page_limit(self, response, check_next_link=True):
+        """ Test that a multi-entry endpoint obeys the page limit.
+
+        Parameters:
+            response (requests.Response): the response to test for page limit
+                compliance.
+
+        Keyword arguments:
+            check_next (bool): whether or not to recursively follow and test any
+                pagination links provided under `links->next`.
+
+        Raises:
+            ResponseError: if test fails in a predictable way.
+
+        Returns:
+            bool, str: True if the test was successful, with a string describing
+                the success.
+
+        """
         try:
             num_entries = len(response.json()["data"])
```

Comment:
```suggestion
            response = response.json()
```

Reply:
Then before the line `if num_entries > self.page_limit` the variable `num_entries` should be defined from `response`.